app/assets/javascripts/switch.js

```diff
@@ -1,5 +1,8 @@
 ready = function() {
-  $('[data-switch]').bootstrapSwitch({ onText: "private", offText: "public" });
+  $.each($('[data-switch]'), function(key, value){
+    console.log(value.getAttribute('data-offText'));
+    $(this).bootstrapSwitch({ offText: value.getAttribute('data-offText'),  onText: value.getAttribute('data-onText')});
```

Comment:
you can use the jquery function data
`$(value).data('offText')`

Doesn't have to, just a suggestion